picard/util/checkupdate.py

```diff
@@ -0,0 +1,177 @@
+)
+import picard.util.webbrowser2 as wb2
+
+# Used to strip leading and trailing text from version string.
+_RE_CLEAN_VERSION = re.compile('^[^0-9]*(.*)[^0-9]*$', re.IGNORECASE)
+
+
+class UpdateCheckManager(QtCore.QObject):
+
+    def __init__(self, parent=None):
+        super().__init__(parent=parent)
+        self._parent = parent
+
+        # PICARD_VERSIONS dictionary valid keys are: 'stable', 'beta' and 'dev'.
+        # Each of these keys contains a dictionary with the keys: 'tag' (string),
+        # 'version' (tuple) and 'urls' (dictionary).  The 'version' tuple comprises
+        # major (int), minor (int), micro (int), type (str) and development (int)
+        # as defined in PEP-440.  The Picard developers have standardized on using
+        # only 'dev' or 'final' as the str_type segment of the version tuple.  Each
+        # key in the 'urls' dictionary contains a string with the specified url.
+        # Valid keys include: 'download' and 'changelog'.  The only required key in
+        # the 'urls' dictionary is 'download'.
+
+        # Initialize empty dictionary for 'stable' key
+        self._available_versions = {
+            'stable': { 'tag': '', 'version': (0, 0, 0, 'dev', 0), 'urls': {'download': ''} },
+        }
+        self._show_always = False
+        self._update_level = 0
+
+    def check_update(self, show_always=False, update_level=0, callback=None):
+        '''Checks if an update is available.
+
+        Compares the version number of the currently running instance of Picard
+        and displays a dialog box informing the user  if an update is available,
+        with an option of opening the download site in their browser.  If there
+        is no update available, no dialog will be shown unless the "show_always"
+        parameter has been set to True.  This allows for silent checking during
+        startup if so configured.
+
+        Args:
+            show_always: Boolean value indicating whether the results dialog
+                should be shown even when there is no update available.
+            update_level: Determines what type of updates to check.  Options are:
+                0 = only stable release versions are checked.
+                1 = stable and beta releases are checked.
+                2 = stable, beta and dev releases are checked.
+
+        Returns:
+            none.
+
+        Raises:
+            none.
+        '''
+        self._show_always = show_always
+        self._update_level = update_level
+
+        if self._available_versions['stable']['tag']:
+            # Release information already acquired from specified website api.
+            self._display_results()
+        else:
+            # Gets list of releases from specified website api.
+            self._query_available_updates(callback=callback)
+
+    @property
+    def available_versions(self):
+        '''Provide a list of the latest version tuples for each update type.'''
+        return self._available_versions
```

Comment:
Is this property used somewhere ?
Why not rename `self._available_versions` to `self.available_versions` instead ?

Reply:
Actually, it's not used anywhere, so I'll just remove it.